index.bs

```diff
@@ -303,6 +284,61 @@ Loader instances are initially created with the internal slots described in the
 
 <h3 id="registry-abstract-operations">Abstract Operations for Registry Objects</h3>
 
+<h4 id="registry-GetCurrentStage" aoid="GetCurrentStage">GetCurrentStage(entry)</h4>
+
+The abstract operation GetCurrentStage with argument entry performs the following steps:
+
+<emu-alg>
+1. If Type(_entry_) is not Object, throw a *TypeError* exception.
+1. Let _stages_ be _entry_.[[Pipeline]].
+1. Return the first element of _stages_.
+</emu-alg>
+
+<h4 id="registry-GetStage" aoid="GetStage">GetStage(entry, stage)</h4>
+
+The abstract operation GetStage with arguments entry and stage performs the following steps:
+
+<emu-alg>
+1. If Type(_entry_) is not Object, throw a *TypeError* exception.
+1. Let _stages_ be _entry_.[[Pipeline]].
+1. For each element _entry_ of _stages_, do
+  1. If _entry_.[[Stage]] is equal to _stage_, then
+    1. Return _entry_.
+1. Return *undefined*.
+</emu-alg>
+
+<h4 id="registry-UpgradeToStage" aoid="UpgradeToStage">UpgradeToStage(entry, stage)</h4>
+
+<emu-alg>
+1. Let _pipeline_ be _entry_.[[Pipeline]].
+1. Let _stageEntry_ be GetStage(_entry_, _stage_).
+1. If _stageEntry_ is not *undefined*, then
+  1. Repeat while the first element of _pipeline_ is not equal to _stageEntry_
+    1. Remove first element from _pipeline_.
+</emu-alg>
+
+An <dfn id="stage-entry">stage</dfn> is a record with the following fields:
+
+<table>
+  <thead>
+    <tr>
+      <th>Internal Slot</th>
+      <th>Value Type (<em>non-normative</em>)</th>
+      <th>Description (<em>non-normative</em>)</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>\[[Stage]]</td>
+    <td><code>"fetch"</code>, <code>"translate"</code>, <code>"instantiate"</code>, <code>"link"</code>, <code>"ready"</code></td>
+    <td>A constant value to indicating which phase the entry is at.</td>
+  </tr>
+  <tr>
+    <td>\[[Result]]</td>
```

Comment:
Question: I think we should use `[[Value]]` instead of `[[Result]]`, it is not really a result, but a thenable value.